packages/react-router-native/index.tsx

```diff
@@ -81,32 +97,34 @@ if (__DEV__) {
  * A <TouchableHighlight> that navigates to a different URL when touched.
  */
 export function Link({
-  as: Component = TouchableHighlight,
   onPress,
   replace = false,
   state,
   to,
   ...rest
-}) {
+}: LinkProps) {
   let navigate = useNavigate();
 
-  return (
-    <Component
-      {...rest}
-      onPress={event => {
-        if (onPress) onPress(event);
-        if (!event.defaultPrevented) {
-          navigate(to, { replace, state });
-        }
-      }}
-    />
-  );
+  function handlePress(event: GestureResponderEvent) {
+    if (onPress) onPress(event);
+    if (!event.defaultPrevented) {
+      navigate(to, { replace, state });
+    }
+  }
+
+  return <TouchableHighlight {...rest} onPress={handlePress} />;
+}
```

Comment:
Here you can type the whole component using the `React.FC` type instead of only typing props.

The advantage is not really noticeable in this specific scenario but it pays off when you have components with multiple return statements as it will throw a TS error if you return something other than `ReactElement | null`.

```tsx
export const Link: FC< LinkProps> = () {
  return <div />;
}
```

This will produce an error:

```tsx
export const Link: FC< LinkProps> = () {
  return undefined;
}
```

Reply:
This is a great suggestion, thanks. But I prefer to use `function` instead of `const` to define my component functions. I'll make sure to add return types explicitly to all my functional components.